src/plugins/data/public/query/state_sync/sync_state_with_url.ts

```diff
@@ -85,7 +85,10 @@ export const syncQueryStateWithUrl = (
     stateContainer: {
       ...globalQueryStateContainer,
       set: state => {
-        globalQueryStateContainer.set(state || defaultState);
+        if (typeof state !== 'object' || state === null) {
+          return;
+        }
+        globalQueryStateContainer.set(state);
```

Comment:
@Dosant when I've clicked the `New` button of the Discover toolbar, the global state was resetted. This adaption prevents this reset, the cause was  a given value of state = `null`

Reply:
Interesting! I've never noticed that button before :D 

I wonder if it actually would be more correct behaviour to reset everything 🤔

Reply:
Just checked 7.6.0. Looks like hitting "new" button removes app filters, but leaves pinned filters. In this branch app filters are preserved.

Reply:
Fixed by using `setAppFilters` like you mentioned in your other comment. Yes, I wonder if there will be a time when one can say about Kibana: I've seen it all. Also don't know if this button is used often. There is no `New` in Visualize and Dashboard, but I think it would be useful.